src/main/java/org/jenkinsci/plugins/credentialsbinding/masking/Base64SecretPatternFactory.java

```diff
@@ -0,0 +1,59 @@
+package org.jenkinsci.plugins.credentialsbinding.masking;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import hudson.Extension;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.Collection;
+import java.util.Collections;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+@Extension
+@Restricted(NoExternalUse.class)
+public class Base64SecretPatternFactory implements SecretPatternFactory {
+    @NonNull
+    @Override
+    public Collection<String> getEncodedForms(@NonNull String input) {
+        return getBase64Forms(input);
+    }
+
+    @NonNull
+    public Collection<String> getBase64Forms(@NonNull String secret) {
+        if (secret.length() == 0) {
+            return Collections.emptyList();
+        }
+
+        Base64.Encoder[] encoders = new Base64.Encoder[]{
+                Base64.getEncoder(),
+                Base64.getUrlEncoder(),
+        };
+
+        Collection<String> result = new ArrayList<>();
```

Comment:
Perhaps use Set over List?

Reply:
Should not matter I think; all the patterns will wind up merged into one big pattern sorted by length.